hydra-node/test/Hydra/Model/MockChain.hs

```diff
@@ -69,82 +85,113 @@ mockChainAndNetwork ::
+                      TxOut
+                        (mkVkAddress testNetworkId vkey)
+                        (lovelaceToValue 10_000_000)
+                        TxOutDatumNone
+                        ReferenceScriptNone
+                 in ScriptRegistry
+                      { initialReference = (txIn, txOut)
+                      , commitReference = (txIn, txOut)
+                      , headReference = (txIn, txOut)
+                      }
+            , contestationPeriod = cp
+            }
+        chainState =
+          S.ChainStateAt
+            { chainState = S.Idle
+            , recordedAt = Nothing
+            }
+    let getTimeHandle = pure $ arbitrary `generateWith` 42
+    let seedInput = genTxIn `generateWith` 42
+    nodeState <- createNodeState $ Idle IdleState{chainState}
+    let HydraNode{eq} = node
+    let callback = chainCallback eq
+    rollbackableChainState <- newRollbackableChainState initialChainState
+    let chainHandle = createMockChain tr ctx (atomically . writeTQueue queue) getTimeHandle seedInput rollbackableChainState
+    let chainHandler = chainSyncHandler tr callback getTimeHandle ctx rollbackableChainState
+    let node' =
+          node
+            { hn = createMockNetwork node nodes
+            , oc = chainHandle
+            , nodeState
+            }
+    let mockNode = MockHydraNode{node = node', chainHandler}
+    atomically $ modifyTVar nodes (mockNode :)
+    pure node'
 
-  simulateTicks queue bumpSlot = forever $ do
-    threadDelay blockTime
+  blockTime :: Integer
+  blockTime = 20 -- seconds
+  simulateTicks queue chain = forever $ do
+    rollForward chain queue
+    rollForward chain queue
+    rollForward chain queue
+    rollForward chain queue
+    sendRollBackward chain 2
+  rollForward chain queue = do
+    threadDelay $ fromIntegral blockTime
+    transactions <- flushQueue queue []
+    addNewBlockToChain chain transactions
+    sendRollForward chain
+  flushQueue queue transactions = do
     hasTx <- atomically $ tryReadTQueue queue
     case hasTx of
       Just tx -> do
-        slotNo <- bumpSlot
-        let header = genBlockHeaderAt slotNo `generateWith` 42
+        flushQueue queue (tx : transactions)
+      Nothing -> pure transactions
+  appendToChain block (slotNum, cursor, blocks) =
+    (slotNum, cursor, blocks :|> block)
+  sendRollForward chain = do
+    (slotNum, position, blocks) <- readTVarIO chain
+    case Seq.lookup position blocks of
+      Just (header, txs) -> do
         allHandlers <- fmap chainHandler <$> readTVarIO nodes
-        forM_ allHandlers $ \ChainSyncHandler{onRollForward} ->
-          onRollForward header [tx]
-      Nothing -> pure ()
+        forM_ allHandlers (\h -> onRollForward h header txs)
+        atomically $ writeTVar chain (slotNum, position + 1, blocks)
+      Nothing ->
+        pure ()
+  sendRollBackward chain nbBlocks =
+    if False -- inhibiting the rollback until we can support it in the code
```

Comment:
plz do not forget to remove this flag

Reply:
Good catch. The mock chain should either do random rollbacks or not. No dead code should remain.